fitbenchmarking/utils/matlab_engine.py

```diff
@@ -0,0 +1,40 @@
+"""
+Manage the persistent variables withing the matlab engine
+"""
+
+import matlab.engine
+ENG = matlab.engine.start_matlab()
+
+
+def add_persistent_matlab_var(name: str) -> None:
+    """
+    Manage a list of variables to keep when clearing memory.
+
+    :param name: The variable to add to the persistent list
+    :type name: str
+    """
+    ENG.evalc(
+        r'if not(exist("persistent_vars"));'
+        r' persistent_vars = {"persistent_vars"};'
+        r'end;'
+    )
+    ENG.evalc(
+        f'if not(any(cellfun(@(x) x=="{name}", persistent_vars)));'
+        f' persistent_vars{{end+1}} = "{name}";'
+        r'end;'
+    )
```

Comment:
Why are the first block raw, but the second block not? Need `rf`?

Reply:
Also 
```
if not(any(cellfun(@(x) x=="{name}", persistent_vars)));
```
Could be
```
if ~ismember("{name}", persistent_vars)
```

Reply:
The raw was so it's single `{` so I dont think rf is useful. Although this probably went through a few iterations and it might be cleaner with the double `{{`.